requirements.txt

```diff
@@ -1,41 +1,2 @@
-colored
-dipy
-# h5py is pinned to minor than 3 due to issues with Keras/TF
-# https://github.com/tensorflow/tensorflow/issues/44467
-h5py~=2.10.0
-Keras==2.3.1
-ivadomed
-matplotlib
-nibabel
-numpy
-# onnxruntime>=1.5.1 requires `brew install libomp` on macOS.
-# So, pin to 1.4.0 to avoid having to ask users to install libomp.
-# ivadomed==2.5.0 would also do this, but #3035 is preventing that.
-onnxruntime==1.4.0
-pandas
-psutil
-pyqt5==5.11.3
-pytest
-pytest-cov
-raven
-requests
-requirements-parser
-scipy
-scikit-image
-scikit-learn
-tensorflow~=1.15.0
-# PyTorch's Linux/Windows distribution is very large due to its GPU support,
-# but we only need that for training models. For users, use the CPU-only version
-# (only available directly from the PyTorch project).
-# The macOS version has never had GPU support, so doesn't need the workaround.
 -f https://download.pytorch.org/whl/cpu/torch_stable.html
```

Comment:
Hmm... I wonder if `requirements.txt` is even necessary anymore if its contents are just:

```
-f https://download.pytorch.org/whl/cpu/torch_stable.html
-e .
```

Since we want to encourage use of `install_sct` (for now), it might be better to get rid of this entirely, so people don't try to run `pip install -r requirements.txt` and end up with a broken installation.

Reply:
Should we include this in another PR (same one that removes `pip` from doc)?

Reply:
I think we could do it in this PR! (And save the documentation update for a separate PR.)

The only other thing we'd have to do, I think, is update `.readthedocs.yml`, which currently relies on the presence of `requirements.txt`:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/e49161416a8e79bc75ce74b55c0ddf5563689327/.readthedocs.yml#L13-L20

I think it's just as simple as going:

```diff
-    - requirements: requirements.txt
```

Since ReadTheDocs already runs `pip install .[docs]` during docs builds (which will install SCT, its dependencies, and the docs `extra_requirements`):

![Screenshot from 2022-01-18 10-38-06](https://user-images.githubusercontent.com/16181459/149968861-91ceb7a1-0470-4a39-b893-78ebb3628adc.png)

Reply:
Ok I tried removing it, but it fails to install torch+cpu for the doc. I couldn't find an argument to include the torch repo unfortunately :(

Reply:
> Ok I tried removing it, but it fails to install torch+cpu for the doc.

_(Linking the build for posterity: https://readthedocs.org/projects/spinalcordtoolbox/builds/15808655/)_

Reply:
Ahh of course, since we can't use `-f` inside `setup.py`, and we're relying on external commands (e.g. `-f` in `install_sct`) to make that work.

Reply:
Perhaps we could open an upstream issue in ReadTheDocs to ask if there's any way they support `-find-links` option for `setup.py`? But, for now, I think we can save this for a follow-up PR, and not delay this PR too much.